meta-refkit/recipes-image/images/refkit-installer-image.bb

```diff
@@ -0,0 +1,270 @@
+            else
+                output_mounted=1
+            fi
+            if ! execute cp -r "$rootfs/boot/EFI_internal_storage" "$output_mountpoint/EFI"; then
+                fatal "copying EFI files failed"
+            fi
+        fi
+        if ! sync; then
+            fatal "syncing data failed"
+        fi
+        cleanup_populate
+    }
+
+    install_image () {
+        input="${INSTALLER_IMAGE_DATADIR}/$CHOSEN_INPUT"
+        output="/dev/$CHOSEN_OUTPUT"
+        info "Installing from $input to $output."
+
+        confirm_install || return 1
+
+        input_mountpoint=
+        input_mounted=
+
+        cleanup_install_image () {
+            [ "$input_mounted" ] && execute umount "$input_mountpoint"
+            [ "$input_mountpoint" ] && rmdir "$input_mountpoint"
+            [ "$input" ] && execute kpartx -d "$input"
+            remove_cleanup cleanup_install_image
+        }
+        add_cleanup cleanup_install_image
+
+        # Assume that there's only one ext4 partition at the end and it
+        # contains the systems' rootfs.
+        loopdev=$(execute kpartx -sav "$input" | tail -1 | sed -e 's/^\(add map \)*\([^ ]*\).*/\2/')
+        if [ ! "$loopdev" ]; then
+            fatal "kpartx failed for $input"
+        fi
+        if ! input_mountpoint=$(mktemp -dt input-rootfs.XXXXXX); then
+            fatal "could not create mount point"
+        fi
+        if ! execute mount "/dev/mapper/$loopdev" "$input_mountpoint"; then
+            fatal "count not mount rootfs from /dev/mapper/$loopdev"
+        else
+            input_mounted=1
+        fi
+
+        # Clear all partition data on disk
+        if ! execute sgdisk -o "$output"; then
+            fatal "sgdisk $output has failed - damaged disk?"
+        fi
+
+        # Read partition description from rootfs.
+        if ! . "$input_mountpoint/boot/emmc-partitions-data"; then
+            fatal "reading $input_mountpoint/boot/emmc-partitions-data failed"
+        fi
+
+        # Create partitions.
+        pnum=0
+        gdisk_pnum=1
+        while [ "$pnum" -lt "$PART_COUNT" ]; do
+            eval size="\$PART_${pnum}_SIZE"
+            eval uuid="\$PART_${pnum}_UUID"
+            eval type_id="\$PART_${pnum}_TYPE"
+            eval lname="\$PART_${pnum}_NAME"
+            eval fs="\$PART_${pnum}_FS"
+
+            if [ "$gdisk_pnum" -eq "$PART_COUNT" ]; then
+                # Make the last partition take the rest of the space
+                if ! execute sgdisk -n "$gdisk_pnum:+0:-1s" -c "$gdisk_pnum:$lname" \
+                       -t "$gdisk_pnum:$type_id" -u "${gdisk_pnum}:${uuid}" -- "$output"; then
+                    fatal "creating rootfs partition failed"
+                fi
+            else
+                if ! execute sgdisk -n "$gdisk_pnum:+0:+${size}M" -c "$gdisk_pnum:$lname" \
+                       -t "$gdisk_pnum:$type_id" -u "${gdisk_pnum}:${uuid}" -- "$output"; then
```

Comment:
I'm bit confused about using both `$gdisk_pnum` and `${gdisk_pnum}` on this line.